pandas/tests/frame/test_constructors.py

```diff
@@ -1205,6 +1205,18 @@ def test_constructor_from_items(self):
                        columns=['one', 'two', 'three'])
         tm.assert_frame_equal(rs, xp)
 
+        # GH 17312
```

Comment:
make this a new tests, also tests ``DataFrame(dict(...))``

Reply:
Not sure what you mean by the second point here. From what I can tell ``from_items`` only works with a list of (key, value) tuples and not a dict? So how do I do a test with a dict?
 Thanks.

Reply:
I was saying, split off these added cases into a new tests.

also test ``DataFrame(dict(....))`` with the same input, its a dict from the tuples (it will yield the  same error messages)

e.g. on current master

```
In [1]: DataFrame.from_items([('A', 1), ('B', 4)])
ValueError: If using all scalar values, you must pass an index

In [3]: DataFrame(dict({'A': 1, 'B': 4}))
ValueError: If using all scalar values, you must pass an index
```